spec/lib/extensions/ar_base_model_spec.rb

```diff
@@ -1,33 +1,38 @@
 RSpec.describe "ar_base_model extension" do
-  context "with a test class" do
-    let(:test_class) do
-      Class.new(ActiveRecord::Base) do
-        def self.name; "TestClass"; end
-      end
-    end
-
-    it ".base_model" do
-      expect(test_class.base_model).to eq(test_class)
+  before do
+    class TestClass < ActiveRecord::Base; end
+    class TestClassSub1 < TestClass; end
+    class TestClassSub2 < TestClass
+      def self.base_model; TestClassSub2; end
```

Comment:
In order to override, I need access to the class object, but this was really annoying and hard to read without actual class constants, so I switched to them instead.